gen/libdir/forbidden.conf

```diff
@@ -45,11 +45,152 @@
+### Hat
+# `Hat` is an a read-read communication edge in `diy7`, namely
+# the end of a thread and the beginning of antoher thread are both read events
+# and they will read the same value. We swith them off by default.
+###
+#-safe [Hat Fre]
+#-safe [Hat DpAddr] [Hat DpData] [Hat DpCtrl]
+#-safe [Hat DMB.SY***] [Hat DMB.LD***]
+#-safe [Hat A PodR*] [Hat Q PodR*]
+#-safe [Hat Amo.Swp] [Hat Amo.Cas] [Hat Amo.LdAdd] [Hat Amo.StAdd]
+#-safe [Hat Amo]
```

Comment:
Why have you disabled these?

Reply:
Sorry is a rebase mistake. I meant to enable them. Thanks for spotting it.